src/Tasks.UnitTests/MSBuild_Tests.cs

```diff
@@ -1524,6 +1524,98 @@ public void CustomTaskWithBuildProjectFilePassesTaskId()
                 File.Delete(projectFile1);
             }
         }
+
+        [Fact]
+        public void InMemoryProject_Build()
```

Comment:
Btw, there are two MSBuild task implementations but I've found tests for only one of them, let me know if there is another place I could add tests for the other implementation as well.